Deployment/asdk-prechecker.ps1

```diff
@@ -0,0 +1,384 @@
+You may also need to update the PowerShell script execution policy with Set-ExecutionPolicy, since the script is not signed : https://technet.microsoft.com/en-us/library/hh849812.aspx 
+
+The Azure Stack Development Kit Pre-Checker script is a PowerShell script published in this public repository so you can make improvements to it by submitting a pull request.
+https://github.com/Azure/AzureStack-Tools
+#>
+
+#requires –runasadministrator
+
+function CheckNestedVirtualization {
+
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking for physical/virtual machine status..."
+
+    $BaseBoard = (Get-WmiObject Win32_BaseBoard)
+    If ($BaseBoard)
+        {
+        If (($BaseBoard.Manufacturer.Tolower() -match 'microsoft' -and $BaseBoard.Product.Tolower() -match 'virtual') -or ($BaseBoard.Manufacturer.Tolower() -match 'vmware'))
+            {
+            write-host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- WARNING : This server seems to be a virtual machine running on Hyper-V or VMware. Running ASDK on a nested hypervisor is not a tested or supported scenario. Setup will not block this, but this but this may lead to performance or reliability issues."
+            $Global:ChecksFailure++
+            }
+            else
+            {
+            write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- This is a physical machine."
+            $Global:ChecksSuccess++
+            }
+        }
+        else
+        {
+        write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- This is a physical machine."
+        $Global:ChecksSuccess++
+        }
+
+}
+
+function CheckInternetAccess {
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking Internet access..."
+
+    # Test AAD http connection.
+    try {
+        $resp = Invoke-WebRequest -Uri "https://login.windows.net" -UseBasicParsing
+        if ($resp.StatusCode -ne 200) {
+            write-host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- Failed to connect to AAD endpoint https://login.windows.net"
+            $Global:ChecksFailure++
+        }
+        else
+        {
+        write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- This machine has internet access (we tried to contact https://login.windows.net)."
+        $Global:ChecksSuccess++
+        }
+    }
+    catch {
+        write-host -ForegroundColor white "["(date -format "HH:mm:ss")"]" $_.Exception.Message
+        write-host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- Failed to connect to AAD endpoint 'https://login.windows.net'."
+        $Global:ChecksFailure++
+    }
+}
+
+function CheckSystemDisk {
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking system disk capacity..."
+
+    $systemDisk = Get-Disk | ? {$_.IsSystem -eq $true}
+    If ($systemDisk.Size -lt 180 * 1024 * 1024 * 1024)
```

Comment:
Currently the system requires 200GB system disk, please update to
`200 * 1024 * 1024 * 1024`

https://docs.microsoft.com/en-us/azure-stack/asdk/asdk-deploy-considerations?view=azs-2008

Reply:
Updated system calculation and error message. Added reference to documentation in the description section at the top of the script.